### PR TITLE
fix(web): Avoid extracting `spinnakerUser` from cert if available in header

### PR DESCRIPTION
**Pull request by ajordens**
Handles a situation where a certificate used for service to service
communication was created with an RFC822 email address.

### CODE REVIEW

File: kork-web/src/main/groovy/com/netflix/spinnaker/filters/AuthenticatedRequestFilter.groovy [hunk 1]
```diff
@@ -115,6 +103,17 @@ class AuthenticatedRequestFilter implements Filter {
       spinnakerRequestId = UUID.randomUUID().toString()
     }
 
+    // only extract from the x509 certificate if `spinnakerUser` has not been supplied as a header
+    if (request.isSecure() && !spinnakerUser) {
```
Comment (cfieber) [inline kork-web/src/main/groovy/com/netflix/spinnaker/filters/AuthenticatedRequestFilter.groovy]: would anything prevent somebody hitting gate with a valid cert but supplying `X-SPINNAKER-USER` to masquerade as someone else in this case?

Comment (ajordens) [reply kork-web/src/main/groovy/com/netflix/spinnaker/filters/AuthenticatedRequestFilter.groovy]: `gate` explicitly does not extract spinnaker headers.

https://github.com/spinnaker/gate/blob/master/gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/GateConfig.groovy#L320

**Comment by iiro** [comment]
Sorry for barging in on here - this discussion just reminded me of an issue from yesterday... I just bumped into an interesting behaviour when using email as a CommonName (when used together with SAML at least): if I did a request with roer using this certificate - and tried to edit any applications' permissions - it would list this CommonName-email address in a permission-dropdown - and remove the other SAML groups which my user does not belong to from the dropdown. Sounds like a bug...? I had to change the CommonName into something else for now - then things are fine. Sounds like ticket? I'll create one of course if...

**Comment by ajordens** [comment]
@iiro This PR shouldn't impact any of the described behavior (which may or may not be a bug --- I'm not too familiar with what roer is doing) ... it's a change in priority in the places we looked for an email address within a certificate or request. 

It specifically affects environments where you've got mTLS enabled between Spinnaker services and enforces that the email address of the authenticated user will be used rather than any email address encoded within the client certificate.

**Pull request merged**